components/TransactionList/index.js

```diff
@@ -1,13 +1,70 @@
+import { Fragment } from "react";
 import TransactionCard from "../TransactionCard";
-import { Ul } from "./TransactionList.styled";
+import * as Styled from "./TransactionList.styled";
+import {
+  findYearsInTransactions,
+  filterTransactionsByYear,
+  filterTransactionsByMonth,
+  sortMonths,
+} from "../../utils/transaction";
+import { months } from "@/utils/months";
 
 function TransactionList({ transactions }) {
+  const yearsInTransactions = findYearsInTransactions(transactions);
+
+  const transactionsPerYearsAndMonths = yearsInTransactions.map(
+    (year, index) => {
+      const transactionsPerYear = filterTransactionsByYear(transactions, year);
+
+      const monthsToMap =
+        index === 0 ? sortMonths(new Date().getMonth()) : months.toReversed();
+
+      const transactionsPerMonths = monthsToMap.map((month) => {
+        const monthIndex = months.indexOf(month);
+        const transactionsPerMonth = filterTransactionsByMonth(
+          transactionsPerYear,
+          monthIndex
+        );
+        if (transactionsPerMonth.length === 0) {
+          return { month, transactions: [] };
+        }
+        transactionsPerMonth.toSorted(
```

Comment:
the toSorted method does not change transactionsPerMonth. So you need to store the returned value in a new constant or chain the methods in line 24.